pybamm/experiment/experiment.py

```diff
@@ -42,13 +42,17 @@ class Experiment:
 
     def __init__(
         self,
-        operating_conditions,
-        period="1 minute",
-        temperature=None,
-        termination=None,
+        operating_conditions: list,
```

Comment:
Could you add another type hint for `operating_conditions` to check if it is a string as requested in https://github.com/pybamm-team/PyBaMM/issues/3623#issue-2043497552? Also, this attribute should have the type hint `list[str]` to indicate that that this accepts a list of strings. Note that using the generic `list` type is for Python 3.9+ only and we support Python 3.8 as well (which has `List`, rather), so we should add

```python
from __future__ import annotations
```

to the top of this file.

Reply:
I believe list[str] is all that is needed for this annotation. I can review and check it more carefully later

Reply:
If `str` works with `list[str]`, then yes (but I think it should be `str | list[str]` though and `Union` for older Python versions). I am not well-versed about type hints tbh

Reply:
A separate `Union` type hint is not required. `from __future__ import annotations` will do the work.

Reply:
I originally suggested just type hints because it is better to have a single interface (list[str]) then try to support multiple interfaces. It makes the code simpler and cuts down on bugs and mistakes. Once type hints are in place on the function, it will show mistakes in the IDE.